openedx/core/djangoapps/schedules/content_highlights.py

```diff
@@ -6,22 +8,28 @@
 
 from xmodule.modulestore.django import modulestore
 
+log = logging.getLogger(__name__)
 
 def course_has_highlights(course_key):
     """
     Does the course have any highlights for any section/week in it?
     This ignores access checks, since highlights may be lurking in currently
     inaccessible content.
     """
-    if not COURSE_UPDATE_WAFFLE_FLAG.is_enabled(course_key):
+    try:
+        course = _get_course_with_highlights(course_key)
+
+    except CourseUpdateDoesNotExist:
         return False
 
-    course = modulestore().get_course(course_key, depth=1)
-    return any(
-        section.highlights
-        for section in course.get_children()
-        if not section.hide_from_toc
-    )
+    else:
+        course_has_highlights = any(
+            section.highlights
+            for section in course.get_children()
+            if not section.hide_from_toc
+        )
+        _log_when_highlights_modal_is_left_empty(course, course_has_highlights)
```

Comment:
A few things:
* the concept of a "modal" is a UI-layer concept and shouldn't be assumed at this layer.
* The value of `course.highlights_enabled_for_messaging` has already been checked by the call to `_get_course_with_highlights` so no need to check it again.  It adds confusion when the business logic is duplicated.
* Given the above, you probably don't need an internal helper for logging - since it's just a single log line.